airbyte-server/src/main/java/io/airbyte/server/handlers/helpers/OAuthPathExtractor.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.server.handlers.helpers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.ArrayList;
+import java.util.List;
+
+public class OAuthPathExtractor {
+
+  private static final String PROPERTIES = "properties";
+  private static final String PATH_IN_CONNECTOR_CONFIG = "path_in_connector_config";
+
+  public static List<List<String>> extractOauthConfigurationPaths(final JsonNode configuration) {
+
+    if (configuration.has(PROPERTIES) && configuration.get(PROPERTIES).isObject()) {
+      final List<List<String>> result = new ArrayList<>();
+
+      configuration.get(PROPERTIES).fields().forEachRemaining(entry -> {
```

Comment:
nit: shouldn't this just be a `forEach`?

Reply:
Yes but it doesn't exist for Iterator :)